tests/unit/components/sl-menu-test.js

```diff
@@ -1,263 +1,518 @@
+        // Select "right" to the first nested sub-menu's first sub-item
+        component.selectRight();
     });
+
+    assert.equal(
+        component.get( 'selectedItem' ),
+        testItems[ 0 ].items[ 0 ].items[ 0 ],
+        "The first nested sub-menu's first sub-item is selected"
+    );
 });
 
-test( 'Menu selection fires proper selection event', function( assert ) {
-    let selectionCounter = 0;
+test( 'selectSubMenu() selects a sub-menu item', function( assert ) {
+    const component = this.subject({
+        items: testItems
+    });
 
-    this.subject({
-        menu: modelStub,
+    assert.equal(
+        component.get( 'selections' ).length,
+        0,
+        'Nothing is selected initially'
+    );
 
-        selectionMade: 'selectionHandler',
+    Ember.run( () => {
+        // Select the first top-level item
+        component.select( 0 );
 
-        targetObject: {
-            selectionHandler() {
-                selectionCounter++;
-            }
-        }
+        // Select the first sub-menu
+        component.selectSubMenu();
+    });
+
+    assert.equal(
+        component.get( 'selectedItem' ),
+        testItems[ 0 ].items[ 0 ],
+        'The first sub-menu item is selected'
+    );
+});
+
+test( 'selectUp() selects an item in the "up" direction', function( assert ) {
+    const component = this.subject({
+        items: testItems
+    });
+
+    assert.equal(
+        component.get( 'selections' ).length,
+        0,
+        'Nothing is selected initially'
+    );
+
+    Ember.run( () => {
+        // Select the first top-level item
+        component.select( 0 );
+
+        // Select the first sub-menu
+        component.selectSubMenu();
+
+        // Select the second sub-menu item
+        component.selectDown();
     });
 
-    const child1 = this.$( 'li:visible' ).first();
-    child1.trigger( 'mouseenter' );
-    child1.trigger( 'click' );
+    assert.equal(
+        component.get( 'selectedItem' ),
+        testItems[ 0 ].items[ 1 ],
+        'The second sub-menu item is selected'
+    );
 
-    const child2 = child1.find( 'li:visible' ).last();
-    child2.trigger( 'mouseenter' );
-    child2.trigger( 'click' );
+    Ember.run( () => {
+        // Select "up" to the first sub-menu item
+        component.selectUp();
+    });
+
+    assert.equal(
+        component.get( 'selectedItem' ),
+        testItems[ 0 ].items[ 0 ],
+        'The first sub-menu item is selected'
+    );
 
-    const child3 = child2.find( 'li:visible' ).first();
-    child3.trigger( 'mouseenter' );
-    child3.trigger( 'click' );
+    Ember.run( () => {
+        // Select "up" again to select the sub-menu's parent
+        component.selectUp();
+    });
 
-    assert.equal( selectionCounter, 3 );
+    assert.equal(
+        component.get( 'selectedItem' ),
+        testItems[ 0 ],
+        'The first top-level item is selected'
+    );
 });
+
+// These tests rely on being able to instantiate the ember-stream/streamService
```

Comment:
I know we previously talked about this and decided to skip this test but now that I'm thinking about things even if we knew how to instantiate and use the `ember-stream/streamService` it would still not be the correct approach in the test to use it.  Conceptually, this may just as well be a database, API or some other service.  The point being, we do not want to test any of it, but rather our interaction with its expected existence.  We should be mocking this service for our tests.  Now the mocking and registering it for usage may still require us to skip the test for now, but if so then we should refine the comment to indicate this rather than what it currently says.